src/hidapi/libusb/hid.c

```diff
@@ -1307,6 +1307,74 @@ static void init_xboxone(libusb_device_handle *device_handle, unsigned short idV
 	}
 }
 
+static bool is_ns2(unsigned short idVendor, unsigned short idProduct)
+{
+	if (idVendor == 0x057e) {
+		return (idProduct == 0x2073);
+	}
+	return false;
+}
+
+static bool ns2_find_bulk_out_endpoint(libusb_device_handle* handle, uint8_t* endpoint_out)
+{
+	struct libusb_config_descriptor* config;
+	if (libusb_get_config_descriptor(libusb_get_device(handle), 0, &config) != 0) {
+		return false;
+	}
+
+	for (int i = 0; i < config->bNumInterfaces; i++) {
+		const struct libusb_interface* iface = &config->interface[i];
+		for (int j = 0; j < iface->num_altsetting; j++) {
+			const struct libusb_interface_descriptor* altsetting = &iface->altsetting[j];
+			if (altsetting->bInterfaceNumber == 1) {
+				for (int k = 0; k < altsetting->bNumEndpoints; k++) {
+					const struct libusb_endpoint_descriptor* ep = &altsetting->endpoint[k];
+					if ((ep->bmAttributes & LIBUSB_TRANSFER_TYPE_MASK) == LIBUSB_TRANSFER_TYPE_BULK && (ep->bEndpointAddress & LIBUSB_ENDPOINT_DIR_MASK) == LIBUSB_ENDPOINT_OUT) {
+						*endpoint_out = ep->bEndpointAddress;
+						libusb_free_config_descriptor(config);
+						return true;
+					}
+				}
+			}
+		}
+	}
+
+	libusb_free_config_descriptor(config);
+	return false;
+}
+
+static void init_ns2(libusb_device_handle *device_handle)
+{
+	const unsigned char DEFAULT_REPORT_DATA[] = {
+		0x03, 0x91, 0x00, 0x0d, 0x00, 0x08,
+		0x00, 0x00, 0x01, 0x00, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF
+	};
+	const unsigned char SET_LED_DATA[] = {
+		0x09, 0x91, 0x00, 0x07, 0x00, 0x08,
+		0x00, 0x00, 0x01, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00
+	};
```

Comment:
Might want to use some enums for things like the first byte (the command) and the second byte (the direction), but that's not needed for the initial merge.